site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,176 @@ Implementors of this API may return a *404* status code in order to indicate tha
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the command-handling adapter instances for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name           | Mandatory | Location                 | AMQP Type | Description |
+| :------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*      | yes       | *properties*             | *string*  | MUST be set to `get-cmd-handling-adapter-instances`. |
+| *gateway_ids*  | yes       | *properties*             | *string*  | A JSON array containing the IDs of the gateways that may act on behalf of the given device. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *get command-handling adapter instances for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}) as well as the properties shown in the following table:
+
+| Name             | Mandatory | Location                 | AMQP Type    | Description |
+| :--------------- | :-------: | :----------------------- | :----------- | :---------- |
+| *content-type*   | no        | *properties*             | *string*     | MUST be set to `application/json` if the invocation of the operation was successful and the body of the response message contains payload as described below. |
+
+The result of a successful invocation is carried in a single Data section of the response message as a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The response message JSON object has the following properties:
+
+| Name                | Mandatory | JSON Type     | Description |
+| :------------------ | :-------: | :------------ | :---------- |
+| *adapter-instances* | *yes*     | *array*       | A non-empty array of JSON objects that represent the command-handling adapter instances along with the device or gateway id. |
+
+Each entry in the *adapter-instances* array has the following properties:
+
+| Name                  | Mandatory | JSON Type     | Description |
+| :-------------------- | :-------: | :------------ | :---------- |
+| *adapter-instance-id* | *yes*     | *string*      | The ID of the protocol adapter instance handling command & control messages for the device given in the *device-id* property. |
+| *device-id*           | *yes*     | *string*      | The ID of the gateway or device that the protocol adapter instance given by the *adapter-instance-id* is handling command & control messages for. |
```

Comment:
Based on the examples given below, this doesn't seem to necessarily be the *device_id* that has been passed in with the request message. A hint would be nice that this is indeed the case and why the client would actually care about this ID.